File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character , Integer> map = new HashMap<>();
        for(char i : s.toCharArray()){
        map.put(i, map.getOrDefault(i, 0) + 1);
        }
        for(char f : t.toCharArray()){
       if(!map.containsKey(f) || map.get(f) <= 0) {
    return false;
} else {
             map.put(f, map.getOrDefault(f, 0) - 1);

        }
        
        }
        
       for(int count : map.values()) {
    if(count != 0) {
        return false;
    }
}
return true;
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = Counter(s)
        for i in t:
            if i not in count:
                return False
            if  count[i] > 1:
                count[i] -= 1
            else:
                del count[i]
        return not count
